buttons/button_base.h

```diff
@@ -24,20 +24,29 @@ class ButtonBase : public Looper,
       while (!DebouncedRead()) YIELD();
       prop.Event(button_, EVENT_PRESSED);
       if (millis() - push_millis_ < 500) {
-        prop.Event(button_, EVENT_DOUBLE_CLICK);
+        doubleclicked_ = true;
```

Comment:
Instead of triggering EVENT_DOUBLE_CLICK now, we flag that a second button press has occurred so that when the button is released, we can evaluate if it is a double-click or a double-click-and-held event.

Reply:
This changes how existing props work.
I think we still need to emit some event here. It can either be EVENT_DOUBLE_CLICK, or it could be something else.
If we use something other than EVENT_DOUBLE_CLICK, we need to do a search-and-replace in existing props to use the new event.

Reply:
It does change how they work in theory, but in practise it seems pretty unnoticeable that double-click events happens after the second button release rather than the second button press.  If we don't supress the double-click event here, the double-click-and-hold doesn't work, as a spurious double-click will always precede it.

Reply:
We can move EVENT_DOUBLE_CLICK to later, but we should still emit *some* event at this point.
Existing props that don't care about double-click-and-hold events can use the event that we emit here instead of the more complicated ones that we emit later.  Maybe the event we emit here should be called EVENT_DOUBLE_PRESSED?

Reply:
done

Reply:
Please also update the existing props to use EVENT_DOUBLE_PRESSED instead of EVENT_DOUBLE_CLICKED.

Reply:
done